zipline/finance/controls.py

```diff
@@ -75,6 +78,16 @@ def fail(self, asset, amount, datetime, metadata=None):
                                       datetime=datetime,
                                       constraint=constraint)
 
+    def log(self, asset, amount, datetime, metadata=None):
+        constraint = repr(self)
+        if metadata:
+            constraint = "{constraint} (Metadata: {metadata})".format(
+                constraint=constraint,
+                metadata=metadata
+            )
+        log.error("Order for %s shares of %s at %s violates trading "
```

Comment:
We don't need to format the string ahead of time here.  Instead, this wants to be:

`log.error('Order for {num_shares} of {asset} at {dt} violates ...', num_shares=amount, asset=asset, dt=datetime ...)`.